### PR TITLE
fix(bytecode): improve analyze_legacy and stack tests

### PR DESCRIPTION
**Pull request by rakita**
## Summary
- Improve `analyze_legacy` padding logic for SWAPN/DUPN/EXCHANGE edge cases using `prev_opcode` tracking
- Simplify the main while loop to only handle PUSH opcodes
- Update stack.rs tests for new Interpreter API

## Test plan
- Existing tests pass including new edge case tests for SWAPN/DUPN/EXCHANGE with STOP

### CODE REVIEW

**Comment by rakita** [comment]
This fixed the inconsistency in padding that immediates would do

File: crates/bytecode/src/legacy/analysis.rs [hunk 1]
```diff
@@ -201,6 +201,13 @@ mod tests {
             assert_eq!(padded_bytecode[0], op);
             assert_eq!(padded_bytecode[1], opcode::STOP);
             assert_eq!(padded_bytecode[2], opcode::STOP);
+
+            // test when there is only opcode with immediate data missing.
+            let bytecode = vec![op];
+            let (_, padded_bytecode) = analyze_legacy(bytecode.into());
+            assert_eq!(padded_bytecode.len(), 2);
+            assert_eq!(padded_bytecode[0], op);
+            assert_eq!(padded_bytecode[1], opcode::STOP);
```
Comment (frangio) [inline crates/bytecode/src/legacy/analysis.rs:210]: Is this right? This corresponds to a single instruction. My understanding was you always want a final STOP instruction. In that case, you'd expect 2 bytes of padding where the first is an immediate and the second is an instruction.
